consensus/consensus.go

```diff
@@ -103,6 +103,8 @@ type Consensus struct {
 	blockHash [32]byte
 	// Block to run consensus on
 	block []byte
+	// BlockHeader to run consensus on
+	blockHeader []byte
```

Comment:
Should reset/clear this data when it's not valid anymore (e.g. ResetState). Otherwise, it may mess things up.

Reply:
reset; I also modify the logic of leader finalizeCommits to make it robust in case of leader change